packages/api/tests/endpoints/granules/test-searchafter-10k.js

```diff
@@ -2,81 +2,90 @@
+  process.env = {
+    ...process.env,
+    ...localStackConnectionEnv,
+    PG_DATABASE: t.context.testDbName,
+  };
+
+  const granulePgModel = new GranulePgModel();
+
+  const collectionName = randomString(5);
+  const collectionVersion = randomString(3);
+  const testPgCollection = fakeCollectionRecordFactory({
+    name: collectionName,
+    version: collectionVersion,
   });
-  t.context.esClient = await getEsClient();
 
-  process.env.auth_mode = 'private';
+  const collectionPgModel = new CollectionPgModel();
+  const collectionPgRecords = await collectionPgModel.create(
+    knex,
+    testPgCollection
+  );
+  // iterate 10k times
+  await pTimes(granuleTotal, ((index) => {
+    if (index % 1000 === 0 && index !== 0) {
+      console.log('Creating granule', index);
+    }
+    const newPgGranule = fakeGranuleRecordFactory({
+      granule_id: randomString(25),
+      collection_cumulus_id: collectionPgRecords[0].cumulus_id,
+    });
+    return granulePgModel.create(knex, newPgGranule);
+  }), { concurrency });
 });
 
 test.after.always(async (t) => {
   delete process.env.auth_mode;
-  await t.context.esClient.client.indices.delete({ index: t.context.esIndex });
+  await destroyLocalTestDb({
+    knex: t.context.knex,
+    knexAdmin: t.context.knexAdmin,
+    testDbName: t.context.testDbName,
+  });
 });
 
-// TODO postgres query doesn't return searchContext
-test.serial.skip('CUMULUS-2930 /GET granules allows searching past 10K results windows with searchContext', async (t) => {
-  const numGranules = 12 * 1000;
-
-  // create granules in batches of 1000
-  for (let i = 0; i < numGranules; i += 1000) {
-    const granules = granuleFactory(1000);
-    // eslint-disable-next-line no-await-in-loop
-    await loadGranules(granules, t);
-    console.log(`${i} of ${numGranules} loaded`);
-  }
-  console.log('Granules loaded.');
-
-  // expect numGranules / 100 loops since the api limit is 100;
-  const expectedLoops = 1 + (numGranules / 100);
-  let actualLoops = 0;
-  let lastResults = [];
-  let queryString = '';
-  let searchContext = '';
-
-  do {
-    actualLoops += 1;
-    // eslint-disable-next-line no-await-in-loop
-    const response = await request(app)
-      .get(`/granules?limit=100${queryString}`)
-      .set('Accept', 'application/json')
-      .expect(200);
-
-    const results = response.body.results;
-    t.notDeepEqual(results, lastResults);
-    lastResults = results;
-
-    searchContext = response.body.meta.searchContext;
-    if (searchContext) {
-      t.is(results.length, 100);
-    } else {
-      t.is(results.length, 0);
-    }
-    queryString = `&searchContext=${response.body.meta.searchContext}`;
-  } while (searchContext !== undefined);
+test.serial('CUMULUS-2930/3967 /GET granules allows searching past 10K results windows using pagination', async (t) => {
+  const response = await request(app)
+    .get('/granules?limit=100&page=101')
+    .set('Accept', 'application/json')
+    .expect(200);
```

Comment:
What would this return if you specified a page that doesn't exist? e.g. `get('/granules?limit=100&page=99999999')`?

Wondering if the API would happily return a successful response for a blank, out-of-range page, or if it'd return something different. If something different, 👍. If there's a possibility you could get a 200 but no results or something, maybe we validate results?

Reply:
If, for example, you ask for page 102, it returns no results. 

I suppose we *could* return something more intelligent on pagination (we're not currently in that endpoint as delivered).       If you think it's worthwhile I can look into what ES does, but I'd expect it's consistent given our implementation strategy.

Reply:
My only concern here is the test itself. We're requesting page 101, which presumably contains results and returns a 200. What if it didn't have results? I... assume... it would return something other than 200?

In other words, is a 200 response indicative of actual results returned?

Reply:
I think the 200 assertion here just is a catch in case the API throws to make the test easier to read if it *does not*.   We could remove it to the same effect, just would make the test more unclear if the API unexpectedly fails.

Reply:
OHHHHHHH ha

Ok, so waaaaaaayyyyy down after all the removals you have the assertion you point out `t.is(response.body.results.length, 1);`. That makes a lot more sense 😆. I didn't see that and thought we were saying the 200 proved that it's returning results.

Ok, I'm good.